packages/client-react-streaming/src/transportedQueryRef.ts

```diff
@@ -164,43 +194,69 @@ function createTransportedQueryRef<
   };
 }
 
-const hydrationCache = new WeakMap<
-  TransportedQueryRef,
-  { cacheKey: CacheKey }
->();
-
 /** @public */
 export function reviveTransportedQueryRef(
   queryRef: TransportedQueryRef,
   client: ApolloClient
 ): asserts queryRef is TransportedQueryRef & ReturnType<typeof wrapQueryRef> {
+  if (unwrapQueryRef(queryRef)) return;
   const {
-    $__apollo_queryRef: { options, stream, queryKey },
+    $__apollo_queryRef: { options, stream },
   } = queryRef;
-  if (!hydrationCache.has(queryRef)) {
-    const hydratedOptions = deserializeOptions(options);
-    const cacheKey: CacheKey = [
-      hydratedOptions.query,
-      canonicalStringify(hydratedOptions.variables),
-      queryKey,
-    ];
-    hydrationCache.set(queryRef, { cacheKey });
-    const internalQueryRef = getSuspenseCache(client).getQueryRef(
-      cacheKey,
-      () =>
-        client.watchQuery({
-          ...hydratedOptions,
-          fetchPolicy: "network-only",
-          context: skipDataTransport(
-            readFromReadableStream(stream.pipeThrough(new JSONDecodeStream()), {
-              ...hydratedOptions.context,
-              queryDeduplication: true,
-            })
-          ),
-        })
+  const hydratedOptions = deserializeOptions(options);
+  const internalQueryRef = getInternalQueryRef(client, hydratedOptions, {
+    ...hydratedOptions,
+    fetchPolicy: "network-only",
+    context: skipDataTransport(
+      readFromReadableStream(stream.pipeThrough(new JSONDecodeStream()), {
+        ...hydratedOptions.context,
+        queryDeduplication: true,
+      })
+    ),
+  });
+  Object.assign(queryRef, wrapQueryRef(internalQueryRef));
+}
+
+function getInternalQueryRef(
+  client: ApolloClient,
+  userOptions: ApolloClient.WatchQueryOptions,
+  initialFetchOptions: ApolloClient.WatchQueryOptions
+) {
+  if (__DEV__) {
+    // this would be a bug in our code, `initialFetchOptions` should always be derived
+    // from `userOptions` and never change `nextFetchPolicy` or we would need additional
+    // logic to handle that case
+    invariant(
+      userOptions.nextFetchPolicy === initialFetchOptions.nextFetchPolicy
```

Comment:
Should we provide a reason here? Perhaps something like:

> "Encountered an unexpected bug in `@apollo/client-react-streaming`. Please file an issue"

I know this should never get hit, but if for some reason an end user encounters this, there will be no explanation to why it crashed.

Reply:
Sure, why not.
613605d0fe1f1184d6aa24a16f97fa634e3ef04b